backend/src/routes/weaviate.ts

```diff
@@ -45,15 +54,61 @@ weaviateRouter.get("/", async (_req, res) => {
 weaviateRouter.get("/test", async (_req, res) => {
   const result = await client.graphql
     .get()
-    .withClassName('ResearchPaper')
+    .withClassName("ResearchPaper")
     .withHybrid({
-      query: 'software',
+      query: "software",
     })
     .withLimit(3)
-    .withFields('title content')
+    .withFields("title content")
     .do();
 
   res.status(201).json({ response: JSON.stringify(result, null, 2) });
 });
 
+weaviateRouter.post("/search", async (req, res) => {
+  const inputString = req.body.text;
+  console.log(inputString);
+
+  if (!inputString) {
+    res.status(400).json({ error: "text is required in request body" });
+  }
+
+  const promptText = `You are an entity extractor now. Extract entities from the following text: '${inputString}'`;
+  try {
+    const response = await axios.post(
+      OPEN_AI_API_URL,
+      {
+        model: "gpt-3.5-turbo",
+        messages: [{ role: "system", content: promptText }],
+        temperature: 0.7,
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${OPEN_AI_APIKEY}`,
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    const entities = response.data.choices[0].message.content.trim();
+    const result = await client.graphql
+      .get()
+      .withClassName("ResearchPaper")
+      .withGenerate({
+        groupedTask: inputString,
```

Comment:
one thingy forgotten here. 

`groupedTask` should be:
`groupedTask: generatePrompt`

and generatePrompt:
```
const generatePrompt =
    "You are an assistant. Answer ONLY with the facts given as a context. \
              Answer shortly for the questions given by the user. \
              Question: {inputString}";
```